core/src/main/java/org/kiwix/kiwixmobile/core/search/SearchActivity.java

```diff
@@ -101,18 +100,16 @@ protected void onCreate(Bundle savedInstanceState) {
   }
 
   public void activateDefaultAdapter() {
-    currentAdapter = defaultAdapter;
-    listView.setAdapter(currentAdapter);
+    recyclerView.setAdapter(defaultAdapter);
   }
 
   public void activateAutoAdapter() {
-    currentAdapter = autoAdapter;
-    listView.setAdapter(currentAdapter);
+    recyclerView.setAdapter(autoAdapter);
   }
 
   @Override
   public void addRecentSearches(List<String> recentSearches) {
-    defaultAdapter.addAll(recentSearches);
+    defaultAdapter.searchList.addAll(recentSearches);
```

Comment:
don't access the fields of another class directly, SearchActivity should not know about `searchList`, make it private and add the method `addAll` to DefaultAdapter